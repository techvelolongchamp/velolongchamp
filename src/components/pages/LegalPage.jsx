import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import ThirdarySection from '../sections/ThirdarySection'
import Layout from '../Layout'
import Header from '../Header'
import Head from '../Head'

import Legali18n from './LegalPage.i18n'

const LegalContainer = styled.article`
  padding-bottom: ${({ theme }) => theme.spacing.xl};

  h3 {
    color: ${({ theme }) => theme.colors.brandTertiary};
    font-size: ${({ theme }) => theme.fonts.large};
    font-weight: 600;
    margin: ${({ theme }) => theme.spacing.l} 0;
  }

  p {
    color: ${({ theme }) => theme.colors.defaultColor};
    font-size: ${({ theme }) => theme.fonts.medium};
    line-height: ${({ theme }) => theme.spacing.l};
    padding: ${({ theme }) => theme.spacing.s} 0;
    text-align: justify;
  }
`

const LegalPage = ({ lng }) => {
  const trad = Legali18n[lng]
  return (
    <Layout lng={lng}>
      <Header noScroll lng={lng} />
      <Head title={trad.pageHeadTitle} url="/legal" lng={lng} />
      <ThirdarySection title={trad.pageTitle} useH1>
        <LegalContainer>
          <h3>{trad.editor.title}</h3>
          <p>{trad.editor.message}</p>
          <h3>{trad.host.title}</h3>
          <p>{trad.host.message}</p>
          <h3>{trad.cookies.title}</h3>
          <p>{trad.cookies.message}</p>
          <h3>{trad.iproperty.title}</h3>
          <p>{trad.iproperty.message}</p>
          <h3>{trad.time.title}</h3>
          <p>{trad.time.message}</p>
          <h3>{trad.law.title}</h3>
          <p>{trad.law.message}</p>
        </LegalContainer>
      </ThirdarySection>
    </Layout>
  )
}

LegalPage.propTypes = {
  lng: PropTypes.string.isRequired,
}

export default LegalPage
