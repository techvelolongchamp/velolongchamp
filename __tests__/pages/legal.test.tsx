import React from 'react'
import { RenderWithProvider as render } from '../provider'

import LegalPage from '../../pages/legal'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

describe('LegalPage unit tests', () => {
  it('should render correctly', () => {
    window.matchMedia = jest.fn().mockImplementation(() => ({
      addListener: jest.fn(),
      removeListener: jest.fn(),
      matches: false,
    }))
    useRouter.mockImplementation(() => ({ locale: 'fr', pathname: '/' }))
    const { container } = render(<LegalPage />)

    expect(container).toMatchSnapshot()
  })
})
