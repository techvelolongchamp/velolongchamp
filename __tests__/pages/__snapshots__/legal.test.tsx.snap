// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegalPage unit tests should render correctly 1`] = `
.c20 {
  background: #f7f7F7;
  border-top: 1px solid #f7f7F7;
  color: #999999;
  padding: 10px 0 5px;
  text-align: center;
}

.c20 a {
  color: #999999;
}

.c20 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c21 {
  background: #0d171b;
  bottom: 0;
  -webkit-transition: all 1s ease;
  transition: all 1s ease;
  color: white;
  font-size: 14px;
  padding: 10px;
  position: fixed;
  text-align: center;
  width: 100vw;
  z-index: 1000;
}

.c21 div {
  margin-top: 10px;
}

.c21 a {
  background: #6d6e71;
  border-radius: 5px;
  color: #fff;
  margin: 0 10px;
  padding: 5px;
}

.c21 button {
  background: #419a1c;
  border-radius: 5px;
  color: #fff;
  margin: 0 10px;
  outline: done;
  padding: 5px;
  width: 40px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100vh;
}

.c4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-top: 80px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: darkorange;
  border-bottom-left-radius: 30px;
  border-top-left-radius: 30px;
  color: white;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 40px;
  font-weight: 100;
  position: fixed;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  right: 0;
  top: 120px;
  z-index: 10;
}

.c1 svg {
  margin: 10px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: darkorange;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  right: -200px;
  top: 120px;
  z-index: 10;
  width: 200px;
  font-size: 16px;
  padding: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  border-bottom-left-radius: 10px;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  height: 30px;
  margin: 10px 20px 0;
  padding: 5px;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: #419a1c;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 120px;
  border-radius: 5px;
}

.c3:hover {
  color: #f7f7F7;
}

.c14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  font-size: 12px;
  margin-left: 5px;
}

.c15 {
  background: none;
  border: none;
  outline: none;
  padding: 5px;
  width: 24px;
}

.c15:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 5;
  background: #f7f7F7;
  padding: 5px 10px;
  border-bottom: 1px solid #fff;
  color: #0d171b;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  margin: 0;
}

.c12 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-left: 1px solid #6d6e71;
  color: #999999;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  height: 30px;
  margin: 0 20px;
  padding-left: 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c12:hover {
  color: #666666;
}

.c13 {
  background: #6d6e71;
  border-radius: 5px;
  color: #fff;
  cursor: pointer;
  font-size: 16px;
  padding: 10px;
}

.c9 {
  height: 70px;
  padding: 5px 0;
}

.c17 {
  margin: 0 auto;
  max-width: 750px;
  padding: 10px;
  height: inherit;
}

.c16 {
  min-height: 600px;
}

.c18 {
  color: #419a1c;
  font-size: 30px;
  margin: 20px 0;
}

.c19 {
  padding-bottom: 30px;
}

.c19 h3 {
  color: #9e9577;
  font-size: 16px;
  font-weight: 600;
  margin: 20px 0;
}

.c19 p {
  color: #0d171b;
  font-size: 14px;
  line-height: 20px;
  padding: 10px 0;
  text-align: justify;
}

@media (max-width:640px) {
  .c4 {
    margin-top: 110px;
  }
}

@media screen and (max-width:640px) {
  .c12 {
    font-size: 14px;
    border: none;
    padding: 0;
    margin: 0 10px;
  }
}

@media screen and (max-width:640px) {
  .c13 {
    font-size: 14px;
  }
}

<div>
  <div
    class="c0"
  >
    <div>
      <div
        class="c1"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.29 3.86L1.82 18a2 2 0 0 0 1.71 3h16.94a2 2 0 0 0 1.71-3L13.71 3.86a2 2 0 0 0-3.42 0z"
          />
          <line
            x1="12"
            x2="12"
            y1="9"
            y2="13"
          />
          <line
            x1="12"
            x2="12.01"
            y1="17"
            y2="17"
          />
        </svg>
      </div>
      <div
        class="c2"
      >
        Signaler un accident ou problème sur l'anneau
        <a
          class="c3"
          data-testid="show-alert"
          href="https://docs.google.com/forms/d/e/1FAIpQLSdmrvUuZ953-3P8NVmNjiG8N78hY1lcdg_hVbr8lnYEqSjM9A/viewform?vc=0&c=0&w=1"
          rel="noopener noreferrer"
          target="_blank"
          title="Formulaire"
        >
          Formulaire
        </a>
      </div>
    </div>
    <main
      class="c4"
    >
      <header
        class="c5"
        id="header"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <a
              class="c8"
              data-testid="title-link"
              href="/"
            >
              <img
                alt="logo"
                class="c9"
                src="/logo.png"
              />
            </a>
          </div>
          <div
            class="c10"
          >
            <nav>
              <ul
                class="c11"
              >
                <li>
                  <a
                    class="c12"
                    href="/blog"
                  >
                    Actualités
                  </a>
                </li>
              </ul>
            </nav>
            <a
              class="c13"
              href="/join"
            >
              Adhérer gratuitement
            </a>
            <div
              class="c14"
            >
              <button
                class="c15"
                data-testid="select-en"
                href="/"
              >
                EN
              </button>
            </div>
          </div>
        </div>
      </header>
      <section
        class="c16"
      >
        <div
          class="c17"
        >
          <h1
            class="c18"
            data-testid="h1-title"
          >
            Mentions légales
          </h1>
          <article
            class="c19"
          >
            <h3>
              Editeur
            </h3>
            <p>
              Vélo Longchamp est une association de loi 1901 dont le siège social
    est situé au :
              <br />
              21 rue Jasmin, 75016 Paris
              <br />
              Président de
    l'association : Gabriel Blanchong
            </p>
            <h3>
              Hébergeur
            </h3>
            <p>
              Les sites www.velo-longchamp.fr et www.velo-longchamp.com sont hébergés par la société ZEIT, Inc, située au 1046 Kearny Street, San Francisco, CA 94133 USA
            </p>
            <h3>
              Cookies
            </h3>
            <p>
              Ce site utilise Google Analytics, un service d’analyse de site internet fourni par Google Inc. (« Google »). Google Analytics utilise des cookies, qui sont des fichiers texte placés sur votre ordinateur, pour aider le site internet à analyser l’utilisation du site par ses utilisateurs. Les données générées par les cookies concernant votre utilisation du site (y compris votre adresse IP) seront transmises et stockées par Google sur des serveurs situés aux Etats-Unis. Google utilisera cette information dans le but d’évaluer votre utilisation du site, de compiler des rapports sur l’activité du site à destination de son éditeur et de fournir d’autres services relatifs à l’activité du site et à l’utilisation d’Internet. Google est susceptible de communiquer ces données à des tiers en cas d’obligation légale ou lorsque ces tiers traitent ces données pour le compte de Google, y compris notamment l’éditeur de ce site. Google ne recoupera pas votre adresse IP avec toute autre donnée détenue par Google. Vous pouvez désactiver l’utilisation de cookies en sélectionnant les paramètres appropriés de votre navigateur. Cependant, une telle désactivation pourrait empêcher l’utilisation de certaines fonctionnalités de ce site. En utilisant ce site internet, vous consentez expressément au traitement de vos données nominatives par Google dans les conditions et pour les finalités décrites ci-dessus.
            </p>
            <h3>
              Propriété intellectuelle
            </h3>
            <p>
              Tous les éléments du site www.velo-longchamp.com et www.velo-longchamp.fr, y compris les documents téléchargeables, sont protégés par le droit d’auteur, des marques ou des brevets. Ils sont la propriété exclusive de l'association Vélo Longchamp. En conséquence, vous ne pouvez en aucun cas et d’aucune manière, reproduire, représenter, diffuser, modifier, concéder tout ou partie de l’un quelconque des éléments reproduits sur le site et tout ou partie du site sans l’accord préalable et express de l'association Vélo Longchamp.
            </p>
            <h3>
              Durée
            </h3>
            <p>
              Les présentes conditions s’appliquent pendant toute la durée de mise en ligne des services offerts par Vélo Longchamp sur le site www.velo-longchamp.com et www.velo-longchamp.fr.
            </p>
            <h3>
              Loi applicable
            </h3>
            <p>
              Les présentes conditions sont soumises à la loi française. L’attribution de compétence en cas de litige, et à défaut d’accord amiable entre les parties, est donnée aux tribunaux français compétents.
            </p>
          </article>
        </div>
      </section>
    </main>
    <footer
      class="c20"
    >
      <p>
        ©Copyright 2019
         - 2021
         - Vélo Longchamp -
         
        Tous droits réservés
      </p>
      <p>
        -
      </p>
      <p>
        <a
          href="/legal"
        >
          Mentions légales
        </a>
      </p>
    </footer>
    <div
      class="c21"
    >
      <p>
        En poursuivant votre navigation sur notre site, vous acceptez l'utilisation de cookies afin de nous permettre d'améliorer votre expérience utilisateur.
      </p>
      <div>
        <button
          data-testid="accept-cookie"
        >
          OK
        </button>
        <a
          href="/legal"
        >
          En savoir plus
        </a>
      </div>
    </div>
  </div>
</div>
`;
